acra/configuring-maintaining/general-configuration/necessary_prep.md

```diff
@@ -0,0 +1,43 @@
+---
+title: Necessary preparations
+bookCollapseSection: true
+weight: 1
+---
+
+# Necessary preparations (keys, ports, etc.)
+
+## Master key
+
+Most Acra services require master key in order to work and perform cryptographic operations.
+It is used, for example, to encrypt other keys in their keyrings.
+These keys are then used in more specific tasks
+(encrypt data in database, hash data for searchable encryption).
+
+The key could be generated with Acra Keymaker:
+```
+$ acra-keys generate --master_key_path /tmp/master_key --keystore=v2
+$ cat /tmp/master_key
+{"encryption":"cgVd5kHock5dy2bBGv6dHmHCgi2Piq3+y43lE7tTgh0=","signature":"fNNgVVBvjaIuL7Ea5lNcs4Ccz8SqLF4bXGhJvLGVL0k="}
+```
+
+This key should then be base64-encoded and passed to Acra services in `ACRA_MASTER_KEY` environment variable.
+
+Like this: `ACRA_MASTER_KEY="$(cat /tmp/master_key | base64)" acra-server ...`
+
+{{< hint warning >}}
+Make sure the master key is stored in a safe place.
+It's more important than any other key.
+{{< /hint >}}
+
+There are other kinds of keys, for example [ones specific for AcraTranslator](../acra_translator/#keys).
```

Comment:
better to use absolute links from the root using `{{< ref "/acra/configuring-maintaining/general-configuration/...." >}}` tag because it will safe us from future problems like copy-paste/moving content with relative links to another place. because after that link will be related to another path and will be broken

Reply:
That doesn't work, maybe a bug, maybe something else. I tried
- `{{< ref "/acra/configuring-maintaining/general-configuration/acra_translator/#keys" >}}`
- `{{< ref "/acra/configuring-maintaining/general-configuration/acra_translator" >}}`

and `hugo` gives error `page not found`.

On the other hand:
- `{{< ref "/acra/configuring-maintaining/general-configuration" >}}` works
- `/acra/configuring-maintaining/general-configuration/acra_translator/#keys` in the web browser opens that exact page I'm pointing to
- so decided to change `../acra_translator/#keys`
  to `/acra/configuring-maintaining/general-configuration/acra_translator/#keys` w/o this hugo magic